src/System.Text.Json/tests/Serialization/WriteValueTests.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class WriteValueTests
+    {
+        [Fact]
+        public static void CanWriteValueToJsonArray()
```

Comment:
We should add a lot more test cases/permutations to catch such cases of interoping with serializer/utf8jsonwriter via the WriteValue API.